stream-chat-android/src/main/java/com/getstream/sdk/chat/view/EndlessScrollListener.kt

```diff
@@ -0,0 +1,45 @@
+package com.getstream.sdk.chat.view
+
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+
+internal class EndlessScrollListener(
+    private val loadMoreListener: OnLoadMoreListener
+) : RecyclerView.OnScrollListener() {
+
+    var paginationEnabled: Boolean = false
+    var loadMoreThreshold: Int = 0
+        set(value) {
+            require(value >= 0) { "Load more threshold must not be negative" }
+            field = value
+        }
+
+    private var scrollStateReset = true
+
+    override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+        if (dy >= 0 || !paginationEnabled) {
+            return
+        }
+        val layoutManager = recyclerView.layoutManager as? LinearLayoutManager
```

Comment:
Shouldn't we just crash if someone tries to use this on a `RecyclerView` that doesn't have a `LinearLayoutManager`? Seems like invalid usage that should be fixed.

Reply:
Added exception if used with inappropriate LayoutManager